src/lib/npm.ts

```diff
@@ -130,90 +128,202 @@ export async function runNpm(args: Omit<ExecPromiseArg, "command">) {
+        versionToInstall,
+        fluenceAquaDependenciesPath,
+      );
+    }
+
+    return [packageNameAndVersionTuple[0], versionToInstall]
+      .filter(Boolean)
+      .join("@");
   }
 
-  const packageNameAndVersionTuple = splitPackageNameAndVersion(
+  return ensurePathInVersionIsCorrect(
     packageNameAndVersion,
+    fluenceAquaDependenciesPath,
+  );
+}
+
+async function updatePackageJSON(fluenceConfig: FluenceConfig) {
+  const fluenceAquaDependenciesPath = await ensureFluenceAquaDependenciesPath();
+
+  const dependenciesWithFixedRelativePath = Object.fromEntries(
+    Object.entries(fluenceConfig.aquaDependencies).map(
+      ([packageName, version]) => {
+        return [
+          packageName,
+          ensurePathInVersionIsCorrect(version, fluenceAquaDependenciesPath),
+        ];
+      },
+    ),
   );
 
-  const [packageName] = packageNameAndVersionTuple;
-  let [, version] = packageNameAndVersionTuple;
+  await writeFile(
+    await getFluenceAquaDependenciesPackageJsonPath(),
+    jsonStringify({ dependencies: dependenciesWithFixedRelativePath }),
+    FS_OPTIONS,
+  );
+}
 
-  if (version === undefined) {
-    version = await getLatestVersionOfNPMDependency(packageName);
+async function getDependenciesFromPackageJSON(parsedPackageJSON: unknown) {
+  if (
+    !hasKey("dependencies", parsedPackageJSON) ||
+    typeof parsedPackageJSON.dependencies !== "object" ||
+    parsedPackageJSON.dependencies === null
+  ) {
+    return null;
   }
 
-  await runNpm({
-    args: ["i", packageNameAndVersion],
-    options: {
-      cwd: await ensureFluenceAquaDependenciesPath(),
-    },
-  });
+  const dependenciesWithVersions = Object.entries(
+    parsedPackageJSON.dependencies,
+  );
 
-  fluenceConfig.dependencies.npm[packageName] = version;
-  await fluenceConfig.$commit();
+  if (
+    !dependenciesWithVersions.every((a): a is [string, string] => {
+      return typeof a[1] === "string";
+    })
+  ) {
+    return null;
+  }
+
+  const fluenceAquaDependenciesPath = await ensureFluenceAquaDependenciesPath();
+
+  return Object.fromEntries(
+    dependenciesWithVersions.map(([packageName, version]) => {
+      return [
+        packageName,
+        ensurePathInVersionIsCorrect(
+          version,
+          fluenceAquaDependenciesPath,
+          true,
+        ),
+      ];
+    }),
+  );
+}
+
+function getFilePath(version: string) {
+  const [, filePath] = [...(version.match(/^file:(.+)$/) ?? [])];
+  return filePath;
+}
+
+function ensurePathInVersionIsCorrect(
+  version: string,
+  fluenceAquaDependenciesPath: string,
+  isBackToFluenceYAML: boolean = false,
+) {
+  const filePath = getFilePath(version);
+
+  if (filePath === undefined || isAbsolute(filePath)) {
+    return version;
+  }
+
+  let [pathA, pathB] = [projectRootDir, fluenceAquaDependenciesPath];
+
+  if (isBackToFluenceYAML) {
+    [pathA, pathB] = [pathB, pathA];
+  }
+
+  return `file:${relative(pathB, resolve(pathA, filePath))}`;
```

Comment:
I'm not sure how this part does work. Can you put a couple of tests for that and/or the explanation? As far as i understand it, the code does the following:
- it gives the relative path from `.fluence/aqua-deps` to fluence project dir plus initial relative path.
.e.g. if i have following structure
`/fluence/src/aqua1/package.json`
`/fluence/src/aqua2/package.json`
and the first one has import `file:../aqua2`
The resulting path would be `file:../../../aqua2`
How will it help to find appropriate package?

Reply:
You understand everything correctly. It's just transforming one relative path to another. I will add a good explanation for this cause it can seem confusing I agree